test/view/iota.cpp

```diff
@@ -50,6 +50,22 @@ void test_iota_minus() {
   CHECK(detail::iota_minus_(I(max), I(max)) == D(0));
 }
 
+template <typename U>
+void test_wrap_around() {
+    using namespace ranges;
+    auto max = std::numeric_limits<U>::max();
+    U from = max - 2;
+    U to = 2;
+
+    auto rng = view::ints(from, to);
+    CHECK(ranges::size(rng) == std::size_t(5));
+    ::check_equal(rng, {U(max - 2), U(max - 1), max, U(0), U(1)});
+
+    auto rng2 = view::closed_ints(from, to);
+    CHECK(ranges::size(rng2) == std::size_t(6));
+    ::check_equal(rng2, {U(max - 2), U(max - 1), max, U(0), U(1), U(2)});
+}
+
```

Comment:
This seems like a perversion to me. And now I feel bad because I didn't stop you before spending time on it. :-( I would prefer a narrow interface on `iota` and `ints`^D^D^D^D`indices` that requires `from < to`.